<launch>

  <arg name="frame_id" default="kinect_rgb_frame"/>

  <include file="$(find state_manager)/launch/stateManager.launch" />
  <include file="$(find pandora_vision_hole)/launch/pandora_vision_hole.launch" />
  <include file="$(find pandora_vision_victim)/launch/victim_node.launch" />

  <include file="$(find pandora_robots_description)/gio/launch/gio_description.launch" />

  <param name="dataset_path" value="/test/benchmark/data/bags_pcl"/>
  <param name="dataset_camera" value="kinect"/>

  <param name="/kinect_rgb_optical_frame/hfov" value="58.0" />
  <param name="/kinect_rgb_optical_frame/vfov" value="45.0" />

  <param name="benchmarkFlag" value="False" />
  <param name="publisherTopic" value="/kinect/depth_registered/points" />
  <param name="publisherMessagePackage" value="sensor_msgs" />
  <param name="publisherMessageType" value="PointCloud2"/>

  <param name="syncSubscriberTopic" value="/pandora_vision/victim_node/processor_log" />
  <param name="subscriberMessagePackage" value="sensor_processor" />
  <param name="subscriberMessageType" value="ProcessorLogInfo"/>

  <param name="subscriberAlertTopic" value="/vision/victim_direction_alert" />
  <param name="subscriberAlertMessagePackage" value="pandora_common_msgs" />
  <param name="subscriberAlertMessageType" value="GeneralAlertVector"/>

  <param name="holeSubscriberTopic" value="/hole_detector/processor_log" />
  <param name="holeSubscriberMessagePackage" value="sensor_processor" />
  <param name="holeSubscriberMessageType" value="ProcessorLogInfo"/>

  <param name="algorithm" value="Victim"/>

  <test test-name="VictimBenchmarkTest" pkg="pandora_vision_victim"
    type="victim_benchmark_test.py" time-limit="20000.0"/>

</launch>
